### PR TITLE
reference-manual: security: add initial doc for boot software updates

### PR DESCRIPTION
**Pull request by igoropaniuk**
Add initial document for boot software updates.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 442fd54f5993bedd7fad2c940b8ebb0359e6f0f4 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/947/mkdocs/artifacts/index.html

**Comment by igoropaniuk** [comment]
work is still in progress, but decided to create PR for existing pieces so anyone who is interested can start reviewing in a casual manner

**Comment by doanac** [comment]
Docs for ac61725e929afd1f8c7816b0994a91271553bc1b are browesable at: https://ci.foundries.io/projects/fio-docs/builds/952/mkdocs/artifacts/index.html

**Comment by igoropaniuk** [comment]
@ricardosalveti @igoropaniuk @Tim-Anderson @vanmaegima @MatthewCroughan @ldts @angolini 
Ready for review. Sorry for my bad english :)

**Comment by doanac** [comment]
Docs for df40cfb17983b7a2da50a1e3480fe0ccb632c66b are browesable at: https://ci.foundries.io/projects/fio-docs/builds/953/mkdocs/artifacts/index.html

**Comment by doanac** [comment]
Docs for b7fb085f8ceb9610cfb37662a05b07dc6eced123 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/954/mkdocs/artifacts/index.html

**Comment by doanac** [comment]
Docs for 1c64a0897ba61a217d93b36dbb613a4ba82a6031 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/955/mkdocs/artifacts/index.html

File: mkdocs/docs/reference-manual/security/boot_software_updates.md [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+2. **boot.cmd** script checks if we’re booting secondary path
+3. In case **upgrade_available** is set we also check if boot firmware upgrade is needed by checking **bootupgrade_available** flag. If both are true, we obtain boot firmware images from newly deployed ostree sysroot and write them to secondary boot path offsets. After that secondary boot bit it set and then warm reset is performed to enforce BootROM to boot secondary boot path
+4. After reboot we boot secondary boot path, so condition from step 2 is not satisfied and we just perform regular boot of Linux.
+5. After Linux is booted ak-lite confirms successful update by clearing **upgrade_available** flag. At this points we’ve validated our new boot firmware images and now we have to flash them to the stable primary path. Additional reboot is needed after this step.
+6.  Regular POR cold reset is performed
+
+## Add new board
+### TF-A/OP-TEE
+TF-A on ARMv8 or OP-TEE on ARMv7, that provide PSCI services to the Linux OS should support **SYSTEM_RESET2** call.
+**SYSTEM_RESET2** implements internal warm reset [Duck Duck Go](https://duckduckgo.com), resetting only CPU.
```
Comment (angolini) [inline mkdocs/docs/reference-manual/security/boot_software_updates.md:97]:
```suggestion
**SYSTEM_RESET2** implements internal warm reset, resetting only CPU.
```

File: mkdocs/docs/reference-manual/security/boot_software_updates.md [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    CONFIG_IMX_WATCHDOG=y
+    CONFIG_SPL_HW_WATCHDOG=y
+    # CONFIG_SPL_WDT is not set
+    CONFIG_SPL_WATCHDOG_SUPPORT=y
+
+### meta-lmp
+#### mfgtool scripts
+
+To deploy boot images to the destination board mfgtools package is used. It uses special configuration
+file with uuu extensions, that contains all needed instructions for correct deployment of boot images.
```
Comment (angolini) [inline mkdocs/docs/reference-manual/security/boot_software_updates.md:152]:
```suggestion
file with UUU extensions, that contains all needed instructions for correct deployment of boot images.
```

**Pull request closed without merging**